client/value.go

```diff
@@ -62,17 +67,16 @@ func (val FieldValue) Bytes() ([]byte, error) {
 	}
 
 	var value any
-	switch tempVal := val.value.(type) {
```

Comment:
thought: This seems very broken and very much looks like a bug to me.  Please let me know if you know it is not a bug.

Reply:
where exactly is do you think there can be a bug? switching from go's type assertion to `NormalValue` type assertion? They are identical.

Reply:
There is very likely a bug in develop branch.

Reply:
I just looked it over - I misread, it (`convertImmutable`) is weirdly written (probably by me...) and does append the nil values instead of setting by index.  Resolving this conversation.